text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,222 @@
+  target into CI, and the CI-related requirements. This review and approval
+  will typically take place in the PR adding the target to CI.
+- In addition, the release team must approve the long-term viability of the
+  target, and the additional work of supporting the target.
+- Tier 1 targets must have substantial, widespread interest within the
+  developer community, and must serve the ongoing needs of multiple production
+  users of Rust across multiple organizations or projects. A tier 1 target may
+  be demoted or removed if it becomes obsolete or no longer meets this
+  requirement.
+- The target must build and pass tests reliably in CI.
+- Building the target and running the testsuite for the target must not take
+  substantially longer than other targets.
+- If running the testsuite requires additional infrastructure (such as systems
+  running the target), the target development team shall arrange to provide
+  such resources to the Rust project, to the satisfaction and approval of the
+  Rust infrastructure team.
+- Tier 1 targets must provide documentation for the Rust community for how to
+  build and run tests for the platform, using emulation if possible, or
+  dedicated hardware if necessary.
+- A tier 1 target may be demoted or removed if it no longer meets these
+  requirements. Any proposal for demotion or removal will be communicated
+  widely to the Rust community, both when initially proposed and before being
+  dropped from a stable release.
+- All tier 2 requirements apply.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+While these criteria attempt to document the policy, that policy still involves
+human judgment. Targets must fulfill the spirit of the requirements as well, as
+determined by the judgment of the approving teams.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The set of approving teams for each tier arose out of discussion with the
+various teams involved with aspects of the Rust project impacted by new
+targets.
+
+
+# Prior art
+[prior-art]: #prior-art
+
+This attempts to formalize and document Rust policy around targets and
+architectures. https://forge.rust-lang.org/release/platform-support.html
+documents some of that policy.
+
+Future expansions of such policy may find requirements from other communities
+useful as examples, such as Debian's [arch
+policy](https://release.debian.org/bullseye/arch_policy.html) and [archive
+criteria](https://ftp-master.debian.org/archive-criteria.html).
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Where, precisely, should we post the official policy?
+- How should we track the maintainers of a target, so that we can page them if
+  we need an issue addressed involving that target? Can we do so using github
+  teams, so that we can mention them in an issue or PR to get their attention?
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
```

Comment:
Is it possible we would ever break the tier structure out into further levels? E.g. could we decide in the future that it's valuable to distinguish between targets based on e.g. whether `std` builds or just `core`, and/or which subsets of the test suite pass?

Additionally, is it worth stating that since this policy only applies to `rustc` as maintained by the Rust teams, there may be some changes or extensions to this policy if there is ever an alternate implementation that gains some traction? E.g., if (purely hypothetically) Rust were someday ISO standardized, that standard could require that conformant implementations provide their own "tier" guarantees following the `rustc` example (even if they only support one platform, they'd need to specify what level of support the implementation has for that platform).

... that would not be relevant until pretty far in the future (if ever), so perhaps it's not worth noting here.

Reply:
I think that whatever tier policy is adopted in this RFC is absolutely something that only applies to `rustc` itself. Whatever a separate project cares to do is up to them.

In terms of a spec, that sounds like a debate for a far off day.

Reply:
This is a Rust RFC for the Rust project's Rust compiler, and I don't think anyone will expect this RFC to set policy for some third-party Rust compiler.

Reply:
@BatmanAoD wrote:
> Is it possible we would ever break the tier structure out into further levels?

It's certainly possible. This RFC isn't proposing to introduce any such additional granularity, but I'll add a note in the future possibilities section.

Reply:
@joshtriplett It seems you've not added the note on breaking the tiers up into further levels. I'd also appreciate a note to the effect of "introducing new tiers is not in the scope of this RFC but is a possibility for future RFCs".